src/Drivers/Paystar/Paystar.php

```diff
@@ -78,14 +78,17 @@ public function purchase()
             'phone' => $details['mobile'] ?? $details['phone'] ?? null,
             'description' => $details['description'] ?? $this->settings->description,
             'callback' => $callback,
-            'sign' =>
-                hash_hmac(
-                    'SHA512',
-                    $amount . '#' . $order_id . '#' . $callback,
-                    $this->settings->signKey
-                ),
         ];
 
+        // signKey is optional
+        if (!empty($this->settings->signKey)) {
+            $data['sign'] = hash_hmac(
+                'SHA512',
+                $amount . '#' . $order_id . '#' . $callback,
```

Comment:
what's the use case of `sign`?

Reply:
You should fill `sign` by `HMAC` and `SHA512`  method by a separate SIGN_KEY (it should provide by user from paystar account) but it's OPTIONAL now and you can remove it without any problem.

Reply:
okay, if it's optional let's not add it! I don't see any requirement to have it.
but if you want to sign it, let's provide `SIGN_KEY` as you said. here I don't see any sign key provided.

Reply:
it's look like old users have SIGN_KEY but for new users not generating. any way maybe old users use SIGN_KEY in projects.

Reply:
there was not any report due to the lack of a sign key! since nobody asked for it, let's not merge this PR. I close it for now, and will open it again if somebody asked about it.